### ISSUE TITLE
Better code snippets - UI & more ...

### ISSUE THREAD

**Issue by vdespa** [a85/POSTMan-Chrome-Extension#9255]
**Is your feature request related to a problem? Please describe.**
The list of code-snippets has not been touched for a long time and this is the first thing many new people coming to Postman see. Writing scripts and tests should be fun and easy. **Postman can help beginners learn how to write API tests in a more intuitive way.**

**Describe the solution you'd like**

- promote the ajv validator instead of the tiny validator (tv)
- with a clear structure, it is easy to add additional script examples
- consider using a more modern JavaScript syntax (like arrow functions, const, let..)
- UI improvement: grouping snippets on categories: variables, tests, general)

Mock

![image](https://user-images.githubusercontent.com/1197635/98559733-46583200-22a7-11eb-8e42-fab02af3c683.png)

**Pre-request tab**

  - General
    - Request body: Convert XML to object _(renamed)_ 
    - Request body: Convert JSON to object _(renamed)_ 
    - Send a request

   - Request manipulation _(NEW!)_ 
     - Add header
     - Modify header
     - Delete header
     - Add cookie
     - Modify cookie
     - Delete cookie 
     - Set request body
     - Set request URL
     - .... 

  - Variables
    - Set a global variable
    - Get a global variable
    - Set an environment variable
    - Get an environment variable
    - Set a collection variable _(currently missing)_ 
    - Get a collection variable _(currently missing)_ 
    - Set a local variable
    - Get a variable

  - Request tests _(NEW!)_ 
    - Request body: JSON value check
    - Request body: contains string
    - Request body: is equal to a string
    - Request header: has header
    - JSON schema validation _(using Ajv)_ 

**Tests tab**

  - General
    - Response body: Convert XML to object _(renamed)_ 
    - Response body: Convert JSON to object _(renamed)_ 
    - Send a request

  - Variables
    - Set a global variable
    - Get a global variable
    - Set an environment variable
    - Get an environment variable
    - Set a collection variable _(currently missing)_ 
    - Get a collection variable _(currently missing)_ 
    - Set a local variable
    - Get a variable

  - Response tests
    - Status code: Code is 200
    - Status code: Name is Created 
    - Response body: JSON value check
    - Resonse body: contains string
    - Resonse body: is equal to a string
    - Response header: has header
    - Resonse time: less than 200ms
    - JSON schema validation _(using Ajv)_

**Comment by a85** [comment]
@vdespa Yep. This hasn't been touched in a long time. We are looking to extend this now through a feature similar to GitHub Gists. Will have some ideas to share soon.

**Comment by skatikia** [comment]
While the good list is long, we have definitely made some progress on the snippets UI. The snippets are now more organised and searchable. Not a long way to go from here…

<img width="1204" alt="Image" src="https://github.com/user-attachments/assets/5adf48b4-cc40-4e45-8a88-ad06eb7ab16d" />

**Comment by vdespa** [comment]
@skatikia that's a step forward, thanks for the update!
